### PR TITLE
Fix deduplication of module fields translations

### PR DESCRIPTION
**Pull request by KinyaElGrande**

### CODE REVIEW

File: server/compose/types/record_detect_duplicates_test.go [hunk 1]
```diff
@@ -355,11 +358,12 @@ func TestDedupRule_checkMultiValueEqualDuplication(t *testing.T) {
 					Set: []RecordValueError{
 						{
 							Kind:    deDupError.String(),
-							Message: locationRule.IssueMultivalueMessage([]string{"{\"coordinates\":[-6.7833479,20.3768206]}", "{\"coordinates\":[0.7833479,10.3768206]}"}),
+                            Message: locationRule.IssueMultivalueMessage(),
 							Meta: map[string]interface{}{
 								"field":         "location",
 								"dupValueField": "location",
 								"rule":          locationRule.String(),
+                                "value": "{\"coordinates\":[-6.7833479,20.3768206]}, {\"coordinates\":[0.7833479,10.3768206]}",
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates_test.go:366]: what are these magic numbers?

File: server/compose/types/record_detect_duplicates_test.go [hunk 2]
```diff
@@ -302,11 +304,12 @@ func TestDedupRule_checkMultiValueEqualDuplication(t *testing.T) {
 					Set: []RecordValueError{
 						{
 							Kind:    deDupError.String(),
-							Message: numberRule.IssueMultivalueMessage([]string{"234", "897"}),
+                            Message: numberRule.IssueMultivalueMessage(),
 							Meta: map[string]interface{}{
 								"field":         "count",
 								"dupValueField": "count",
 								"rule":          numberRule.String(),
+                                "value": "234, 897",
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates_test.go:312]: leftover?

**Review by petergrlica — CHANGES_REQUESTED** [review]

File: server/compose/types/record_detect_duplicates_test.go (hunk 1 above)
Comment (KinyaElGrande) [reply server/compose/types/record_detect_duplicates_test.go:366]: They are just test data co-ordinates

**Comment by katrinDY** [comment]
LGTM

File: server/compose/types/record_detect_duplicates_test.go (hunk 1 above)
Comment (petergrlica) [reply server/compose/types/record_detect_duplicates_test.go:366]: sorry, mistakenly did not see it was a test file

File: server/compose/types/record_detect_duplicates_test.go (hunk 2 above)
Comment (petergrlica) [reply server/compose/types/record_detect_duplicates_test.go:312]: sorry, mistakenly did not see it was a test file

File: server/compose/types/record_detect_duplicates.go [hunk 3]
```diff
@@ -160,49 +160,52 @@ func (rule DeDupRule) checkDuplication(ctx context.Context, ls localeService, re
         moduleField := rec.module.Fields.FindByName(c.Attribute)
 
         if moduleField.Multi && c.IsAllEqual() {
-            return rule.multiValueAllEqual(ctx, ls, c, rvv, existingVv)
-        }
+            multiValErr := rule.multiValueAllEqual(ctx, ls, c, rvv, existingVv)
+            if out == nil {
+                out = &RecordValueErrorSet{}
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates.go]: move this to the top of function so you won't need to check for == nil

File: server/compose/types/record_detect_duplicates.go [hunk 4]
```diff
@@ -160,49 +160,52 @@ func (rule DeDupRule) checkDuplication(ctx context.Context, ls localeService, re
...
-                    if out == nil {
-                        out = &RecordValueErrorSet{}
+            out.Push(multiValErr)
+        } else {
+            _ = vv.Walk(func(v *RecordValue) error {
+                if v.RecordID != rec.ID {
+                    _ = rvv.Walk(func(rv *RecordValue) error {
+                        if len(rv.Value) > 0 && matchValue(c.Modifier, rv.Value, v.Value) {
+                            valErr.Push(RecordValueError{
+                                Kind:    rule.IssueKind(),
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates.go]: can we move this out of the loop?

File: server/compose/types/record_detect_duplicates.go [hunk 5]
```diff
@@ -160,49 +160,52 @@ func (rule DeDupRule) checkDuplication(ctx context.Context, ls localeService, re
...
+                                Message: ls.T(ctx, "compose", rule.IssueMessage()),
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates.go]: same same

File: server/compose/types/record_detect_duplicates.go [hunk 6]
```diff
@@ -160,49 +160,52 @@ func (rule DeDupRule) checkDuplication(ctx context.Context, ls localeService, re
...
+                        }
+                        return nil
+                    })
+
+                    // 1. multiValue is empty, then all value needs to be a match then return error/warning
+                    // 2. multiValue is oneOf, then one or more value needs to be a match then return error/warning
+                    // 3. multiValue is equal, then all value needs to be a match then return error/warning
+                    if (!valErr.IsValid() && (!c.HasMultiValue() || c.IsAllEqual()) && valErr.Len() == rvv.Len()) || (c.IsOneOf() && valErr.Len() > 0) {
+                        if out == nil {
+                            out = &RecordValueErrorSet{}
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates.go]: no need for this if at the top of function

File: server/compose/types/record_detect_duplicates.go [hunk 7]
```diff
@@ -160,49 +160,52 @@ func (rule DeDupRule) checkDuplication(ctx context.Context, ls localeService, re
...
 }
 
-func (rule DeDupRule) multiValueAllEqual(ctx context.Context, ls localeService, c *DeDupRuleConstraint, rvv RecordValueSet, existingVv RecordValueSet) (out *RecordValueErrorSet) {
-    var (
-        valErr = &RecordValueErrorSet{}
-    )
+func (rule DeDupRule) multiValueAllEqual(ctx context.Context, ls localeService, c *DeDupRuleConstraint, rvv RecordValueSet, existingVv RecordValueSet) (out RecordValueError) {
+    var valErr = RecordValueError{}
 
     if rvv.Len() == existingVv.Len() {
```
Comment (petergrlica) [inline server/compose/types/record_detect_duplicates.go]: you can make an early return with `if rvv.Len() != existingVv.Len()` so you wont have so much nesting

**Pull request merged**